zeppelin-zengine/src/main/java/org/apache/zeppelin/storage/LocalConfigStorage.java

```diff
@@ -95,17 +99,29 @@ public String loadCredentials() throws IOException {
   @Override
   public void saveCredentials(String credentials) throws IOException {
     LOGGER.info("Save Credentials to file: " + credentialPath);
-    writeToFile(credentials, credentialPath);
+    atomicWriteToFile(credentials, credentialPath);
   }
 
   private String readFromFile(File file) throws IOException {
     return IOUtils.toString(new FileInputStream(file));
   }
 
-  private void writeToFile(String content, File file) throws IOException {
-    FileOutputStream out = new FileOutputStream(file);
-    IOUtils.write(content, out);
+  private void atomicWriteToFile(String content, File file) throws IOException {
+    File directory = file.getParentFile();
+    String fileNameParts[] = file.getName().split("\\.");
```

Comment:
isn't this split on `\` platform specific?

Reply:
The original idea was to split on the dot separator between the filename and the `json` type at the end of the name. It turns out that this separation (hence _split_) is not necessary as the following invocation of `File.createTempFile(...)` is perfectly happy with the entire filename (including separator and type) passed in one chunk.